### PR TITLE
Add ReferenceItemsPredictor

### PR DESCRIPTION
**Pull request by dfederm**

### CODE REVIEW

File: src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+{
+    using System;
+    using System.IO;
+    using Microsoft.Build.Evaluation;
+    using Microsoft.Build.Execution;
+
+    /// <summary>
+    /// Finds EmbeddedResource items as inputs.
+    /// </summary>
+    public class ReferenceItemsPredictor : IProjectPredictor
```
Comment (erikma) [inline src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs]: >ReferenceItemsPredictor [](start = 17, length = 23)

Not added to top-level predictor list and UT

File: src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                string hintPath = item.GetMetadataValue(HintPathMetadata);
+                if (!string.IsNullOrEmpty(hintPath))
+                {
+                    predictionReporter.ReportInputFile(hintPath);
+                    continue;
+                }
+                else
+                {
+                    // If there is no hint path then if the reference is valid then the EvaluatedInclude is either a path to a file
+                    // of the name of a dll from the GAC or platform.
```
Comment (erikma) [inline src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs]: >of [](start = 23, length = 2)

or

File: src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    /// <summary>
+    /// Finds EmbeddedResource items as inputs.
+    /// </summary>
+    public class ReferenceItemsPredictor : IProjectPredictor
+    {
+        internal const string ReferenceItemName = "Reference";
+
+        internal const string HintPathMetadata = "HintPath";
+
+        private readonly char[] _invalidPathCharacters = Path.GetInvalidPathChars();
```
Comment (erikma) [inline src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs]: >  [](start = 15, length = 1)

static

File: src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs (hunk 1 above)
Comment (dfederm) [reply src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs]: Damnit! Maybe I do need a reflection UT for that...

File: src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs (hunk 3 above)
Comment (dfederm) [reply src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs]: Will add comment as to why this isn't static

**Pull request merged**